web/app.js

```diff
@@ -1542,6 +1542,68 @@ const PDFViewerApplication = {
     };
     internalEvents.set("updatefromsandbox", updateFromSandbox);
 
+    const lastOpenPages = new Map();
```

Comment:
Can you add a comment explaining this section. The async function with double nested promises is rather confusing.

Reply:
Please see https://github.com/mozilla/pdf.js/pull/12816#discussion_r552157861, where I've added a suggestion that removes the need for the added complexity the the `lastOpenPages` brings.